go-controller/pkg/ovn/address_set.go

```diff
@@ -89,7 +107,36 @@ func (asf *ovnAddressSetFactory) ForEachAddressSet(iteratorFn AddressSetIterFunc
 	return nil
 }
 
+func truncateSuffixFromAddressSet(asName string) string {
+	// Legacy address set names will not have v4 or v6 suffixes.
+	// truncate them for the new ones
+	if strings.HasSuffix(asName, ipv4AddressSetSuffix) {
+		return strings.TrimSuffix(asName, ipv4AddressSetSuffix)
+	}
+	if strings.HasSuffix(asName, ipv6AddressSetSuffix) {
+		return strings.TrimSuffix(asName, ipv6AddressSetSuffix)
+	}
+	return asName
+}
+
 func (asf *ovnAddressSetFactory) DestroyAddressSetInBackingStore(name string) error {
+	// We need to handle both legacy and new address sets in this method. Legacy names
+	// will not have v4 and v6 suffix as they were same as namespace name. Hence we will always try to destroy
+	// the address set with raw name(namespace name), v4 name and v6 name.  The method destroyAddressSet uses
+	// --if-exists parameter which will take care of deleting the address set only if it exists.
+	err := destroyAddressSet(name)
+	if err != nil {
+		return err
+	}
+	err = destroyAddressSet(getIPv4ASNameForNamespace(name))
```

Comment:
comment from before still applies; `name` is not necessarily a _namespace_ name; it might be the name of an address set associated with a NetworkPolicy instead. So the function should not refer to `ForNamespace`

Reply:
Done